client/controllers/contacts.spec.js

```diff
@@ -0,0 +1,45 @@
+describe("The client-side ContactController", function() {
+    // module() is provided by angular-mocks.js
+    beforeEach(module("app"));
+    
+    var $controller, $httpBackend, $rootScope;
+    var contactsController, expectedAllContacts, expectedNewContact;
+    
+    // $controller is the service responsible for instantiating controllers.
+    // It is not available on the global scope, so we must inject it this way.
+    beforeEach(inject(function($injector) {
+        $httpBackend = $injector.get("$httpBackend");
+        $rootScope = $injector.get("$rootScope");
+        $controller = $injector.get("$controller");
+        
+        expectedAllContacts = readJSON("development-data.json").allContacts;
+        expectedNewContact = readJSON("development-data.json").newContact;
```

Comment:
This probably doesn't need to be reset before each test. Move this to `it can get all contacts`.